### PR TITLE
add toTextTable() function

### PR DESCRIPTION
**Pull request by 666666fzl**
add toTextTable() method in Table.java, so that it produces text-formatted table

### CODE REVIEW

**Comment by mssammon** [comment]
@666666fzl your PR is failing the CI build.  It looks like a license header issue.  Are you able to see the details of the semaphoreci failure?

**Comment by 666666fzl** [comment]
@mssammon This is the same problem I talked about when I tried to build. Later I worked around by importing into IntelliJ. Here's the error msg:

> Downloading: https://repo.maven.apache.org/maven2/edu/illinois/cs/cogcomp/illinois-common-resources/1.3/illinois-common-resources-1.3.jar
>                    
> [INFO] ------------------------------------------------------------------------
> [INFO] Reactor Summary:
> [INFO] 
> [INFO] illinois-cogcomp-nlp ............................... SUCCESS [  0.526 s]
> [INFO] illinois-core-utilities ............................ SUCCESS [  4.111 s]
> [INFO] illinois-curator ................................... SUCCESS [  2.530 s]
> [INFO] illinois-tokenizer ................................. SUCCESS [  1.388 s]
> [INFO] illinois-corpusreaders ............................. SUCCESS [  1.466 s]
> [INFO] illinois-edison .................................... FAILURE [03:09 min]
> [INFO] illinois-lemmatizer ................................ SKIPPED
> [INFO] illinois-pos ....................................... SKIPPED
> [INFO] illinois-chunker ................................... SKIPPED
> [INFO] illinois-ner ....................................... SKIPPED
> [INFO] LBJava-NLP-tools ................................... SKIPPED
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 03:19 min
> [INFO] Finished at: 2016-09-19T20:22:19+00:00
> [INFO] Final Memory: 54M/724M
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal on project illinois-edison: Could not resolve dependencies for project edu.illinois.cs.cogcomp:illinois-edison:jar:3.0.72: Could not transfer artifact edu.illinois.cs.cogcomp:illinois-common-resources:jar:1.3 from/to CogcompSoftware (http://cogcomp.cs.illinois.edu/m2repo/): GET request of: edu/illinois/cs/cogcomp/illinois-common-resources/1.3/illinois-common-resources-1.3.jar from CogcompSoftware failed: Premature end of Content-Length delimited message body (expected: 457574985; received: 227851024 -> [Help 1]
> [ERROR] 
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR] 
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
> [ERROR] 
> [ERROR] After correcting the problems, you can resume the build with the command
> [ERROR]   mvn <goals> -rf :illinois-edison

**Comment by mssammon** [comment]
@666666fzl please also add a reasonably descriptive PR description -- you can edit your original PR comment.

**Comment by mssammon** [comment]
@bhargav this addresses https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/issues/49, which you opened, so I'm assigning you to review...

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/Table.java [hunk 1]
```diff
@@ -28,6 +29,75 @@
...
+                    currTableRow.add(this.getValueAt(row, i/2).toString());
+                }
+            }
+            allTableRows.add(currTableRow);
+        }
+
+        // Change the space between '|' if the longest word has length longer than 10(default)
+        if (longestWordSize > 10) {
+            for (int i = 1; i < formatterList.size(); i += 2) {
+                formatterList.set(i, "%-" + longestWordSize + "s ");
```
Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/Table.java]: You should add to the documentation that all columns are of equal size = Max(10, longest word)

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/Table.java [hunk 2]
```diff
@@ -28,6 +29,75 @@
...
+
+        String formatterString = "";
+        for (int i = 0; i < formatterList.size(); i ++) {
+            formatterString += formatterList.get(i);
+        }
+
+        for (int i = 0; i < allTableRows.size(); i ++) {
+            // Add this line to separate data from their column names
+            if (i == 1) {
+                buffer.append("|-----\n");
```
Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/Table.java]: You can have this separate extend till the end of all columns. (We have the longestWordSize and number of columns)

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/Table.java [hunk 3]
```diff
@@ -114,4 +184,20 @@ public String getFormattedString(double d, int numDecimalPlaces) {
     public void addSeparator() {
         separators.add(this.getRowCount());
     }
+
+    public static void main(String []args) {
```
Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/Table.java]: Can you remove the main method? 
If you want, you can move some of the logic under the test file as a unit test. (Though its fine to not have unit tests for table formatting).

**Comment by bhargav** [comment]
@666666fzl Added some minor comments.

**Comment by 666666fzl** [comment]
@bhargav Hi Bhargav, can you take a look at my update? thx

**Comment by bhargav** [comment]
@666666fzl Looks good to me. The Semaphore CI is failing due to an unrelated issue. I'll check with Daniel/Mark tomorrow before merging this.

**Comment by danyaljj** [comment]
@666666fzl could you update your branch with master?

**Comment by 666666fzl** [comment]
@danyaljj Hi Daniel, is it good now?

**Pull request merged**